yams-py/yams-py/Patcher.py

```diff
@@ -0,0 +1,79 @@
+import json
+import os
+import platform
+import shutil
+import subprocess
+import sys
+import tempfile
+from pathlib import Path
+
+# TODO: revise this.
+yams_path = os.fspath(Path(__file__).with_name(name="yams"))
+sys.path.append(yams_path)
+from pythonnet import load
+load("coreclr")
+import clr
+clr.AddReference("YAMS-LIB")
+from YAMS_LIB import Patcher
+
+
+def patch_game(input_path, output_path, patch_data):
+    # Copy to input dir to temp dir first to do operations there
+    progress_update("Copying to temporary path...", -1)
+    tempdir = tempfile.TemporaryDirectory()
+    shutil.copytree(input_path, tempdir.name, dirs_exist_ok=True)
+
+    # Get data.win path. Both of these *need* to be strings, as otherwise patcher won't accept them.
+    output_data_win_path: str = os.fspath(self._get_data_win_path(tempdir.name))
+    input_data_win_path: str = shutil.move(output_data_win_path, output_data_win_path + "_orig")
+
+    # Temp write patch_data into json file for yams later
+    progress_update("Creating json file...", -1)
+    json_file = tempfile.NamedTemporaryFile(mode='w+', delete=False)
```

Comment:
Just keep the file there. You'll be grateful later on when people report bugs that there's an easy way to grab this one.

Reply:
The file isnt actually deleted right now though?

Reply:
Then why you're using a temporary file? And as a generated temporary file you won't be easily finding this later.
Just put it next to the game data with a static name.

Reply:
Because I *was* planning to delete the file originally, i just forgot to write the line xP